### PR TITLE
Fixed issue #33 - Backend Authentification.

### PR DESCRIPTION
**Pull request by LouGel**
Created JWT based authentification.
Auth and role controller.
Database stored Users.
Create users , getUser foncitons.
Swagger doc.
Put previous apis public.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -17,6 +18,7 @@ async function bootstrap() {
     .setVersion('1.0')
     .addTag('CCPM')
     .build();
+  await bootstrapAuthAndUsers();
```
Comment (MartianGreed) [inline src/main.ts:21]: this should be setup in test files or fixtures

File: src/users/user.controller.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { Controller, Get, Param } from '@nestjs/common';
```
Comment (MartianGreed) [inline src/users/user.controller.ts:1]: seems those api routes wont be used for the moment. I think we should remove this

File: src/app.module.ts [hunk 3]
```diff
@@ -18,9 +23,22 @@ import { ConsoleModule } from './console/console.module';
...
+  ],
+  providers: [
+    {
+      provide: APP_GUARD,
+      useClass: AuthGuard,
+    },
+    {
+      provide: APP_GUARD,
+      useClass: RolesGuard,
+    },
```
Comment (MartianGreed) [inline src/app.module.ts:41]: i think this duplicates `@Injectable` annotations

**Review by MartianGreed — COMMENTED** [review]
we are lacking tests around services other than that this seems ok to me ! Great job !

File: src/users/user.controller.ts (hunk 2 above)
Comment (LouGel) [reply src/users/user.controller.ts:1]: Okay , maybe i'll put the create user in this route instead. It's more logic. But i'll remove the getters.

File: src/app.module.ts (hunk 3 above)
Comment (LouGel) [reply src/app.module.ts:41]: I wanted to put this  decorators globally (even with the graphQl requests). Dev will have to explicitly add the @Public() deocrator if they want their api to be public.

**Review by MartianGreed — APPROVED** [review]

**Review by schwepps — APPROVED** [review]

**Comment by MartianGreed** [comment]
> I pushed my test commented. Need env.test variables. Need to regenerate prisma model. An reset db. Before running it.

Sure I'll do it this week !

**Comment by LouGel** [comment]
Hello , there were an issue with jest-config. it included all ts files. even its own config so i restricted it to src and avoid spec test redundancy.

**Pull request merged**